service/history/ndc/history_replicator.go

```diff
@@ -840,6 +840,12 @@ func (r *HistoryReplicatorImpl) notify(
 	clusterName string,
 	now time.Time,
 ) {
+	if len(clusterName) == 0 {
+		// The current cluster does not know about a remote cluster metadata and this can receive an empty cluster.
+		// Ignore this case.
+		return
+	}
```

Comment:
I guess this case can be removed then as source cluster won't generate replication task with empty source cluster?